doc/whats_new.rst

```diff
@@ -253,6 +253,10 @@ Enhancements
 
    - Added :func:`metrics.pairwise.laplacian_kernel`.  By `Clyde Fare <https://github.com/Clyde-fare>`_.
 
+   - Added ``algorithm="elkan"`` to :class:`cluster.KMeans` implementing
+     Elkan's fast K-Means algorithm. By `Andreas Müller`_.
```

Comment:
I just picked up @amueller 's code, I am not at all familiar with what's happening here.